### PR TITLE
generate llc test framework

### PR DESCRIPTION
**Pull request by dw511214992**
This PR also deletes unused files in test by adding them to .gitignore

### CODE REVIEW

File: src/restLevelClient/generateRestLevel.ts [hunk 1]
```diff
@@ -43,6 +48,13 @@ export async function generateRestLevelClient() {
   generateLicenseFile(project);
   generateTsConfig(project);
   generateApiExtractorConfig(project);
+
+  generateKarmaConfigFile(project)
+  generateEnvFile(project);
+  generateEnvBrowserFile(project);
+  generateRecordedClientFile(project);
+  generateSampleTestFile(project);
+
```
Comment (qiaozha) [inline src/restLevelClient/generateRestLevel.ts:57]: add a if generateTest is set then generate those logic here ?

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
         "integration-test:new": "npm-run-all start-test-server generate-and-test integration-test:alone stop-test-server",
         "integration-test:browser": "npm run build:test:browser && karma start karma.conf.js",
         "generate-and-test": "npm-run-all -s build generate-swaggers integration-test:alone integration-test:browser",
-        "integration-test:alone": "mocha -r ts-node/register --timeout 2000 './test/integration/**/*spec.ts'",
+        "integration-test:alone": "mocha -r ts-node/register --timeout 2000 './test/integration/**/!(sampleTest).spec.ts'",
```
Comment (qiaozha) [inline package.json:18]: why do we change the integration-test script in autorest.typescript repo's package.json here ?

Comment (qiaozha) [reply package.json:18]: Got it

File: .gitignore [hunk 3]
```diff
@@ -3,9 +3,10 @@ package-lock.json
 node_modules
 test.json
 test/**/dist
+test/**/dist-test
```
Comment (qiaozha) [inline .gitignore:6]: can we also add the ignore logic for `test/**/types` and `test/**/temp`

File: src/restLevelClient/generateRestLevel.ts (hunk 1 above)
Comment (dw511214992) [reply src/restLevelClient/generateRestLevel.ts:57]: there is if in functions

**Review by qiaozha — APPROVED** [review]

**Comment by dw511214992** [comment]
@joheredi  Although this PR is merged, you still can add some comments if you have any concerns. I will check it and fix them. Thanks

**Pull request merged**